R/expect_not_linter.R

```diff
@@ -11,8 +11,8 @@
 #' @export
 expect_not_linter <- function() {
   xpath <- "//expr[
-    SYMBOL_FUNCTION_CALL[text() = 'expect_true' or text() = 'expect_false']
-    and following-sibling::expr[1][OP-EXCLAMATION]
+    expr[1][SYMBOL_FUNCTION_CALL[text() = 'expect_true' or text() = 'expect_false']]
+    and expr[2][OP-EXCLAMATION]
```

Comment:
This actually changes the matched expr node from the function call node (i.e. only the function symbol) to its parent node (i.e. the entire call).

Is this intentional?

Reply:
yep -- did it for consistency with the other linters.

Also just double checked, and the marker is definitely superior:

```
# on main
lint("\nexpect_true(!A)", expect_not_linter())
<text>:2:1: warning: expect_false(x) is better than expect_true(!x), and vice versa.
expect_true(!A)
^~~~~~~~~~~

# this branch
<text>:2:1: warning: expect_false(x) is better than expect_true(!x), and vice versa.
expect_true(!A)
^~~~~~~~~~~~~~~
```

If anything, I'd move it to put the marker on the `!A` part... for another day/PR